app/models/invitation.rb

```diff
@@ -67,15 +67,21 @@ def send_and_set_date(resend: false)
       UserMailer.invitation(self.id).deliver_now
     end
 
-    date_column = resend ? :resent_at : :sent_at
     # Skip callbacks within after_save by using update_column to avoid a callback loop
-    self.update_column(date_column, Time.current)
+    if resend
+      self.update_column(:resent_at, Time.current)
+      # This applies to old invites when AO3-6094 wasn't fixed.
+      self.update_column(:sent_at, self.updated_at) if self.sent_at.nil?
```

Comment:
Does anyone have strong feelings about calling `update_column` twice? I'm not sure how I feel about it.

Reply:
I think I'm OK with it, the performance difference doesn't seem too bad. But if we want to avoid it, something like
```rb
attrs = { resent_at: Time.current }
attrs[:sent_at] = self.updated_at if self.sent_at.nil?
self.update_columns(attrs)
```
should work